src/Util/MultiSig.hs

```diff
@@ -160,19 +157,20 @@ mergePackages (p :| ps) = maybeToRight PackageMergeFailure $
   foldM mergeSignatures p ps
 
 getBytesToSign :: Package -> Text
-getBytesToSign Package{..} = addTezosBytesPrefix pkToSign
+getBytesToSign Package{..} =
+  addTezosBytesPrefix $ encodeHex $ unHexJSONByteString pkToSign
+  where
+    addTezosBytesPrefix :: Text -> Text
+    addTezosBytesPrefix = ("0x" <>)
 
 -- | Extract the signable component from package. This is a structure
 -- with the packed parameter that represent the action, the multi-sig contract
 -- address, and the replay attack prevention counter.
 getToSign :: Package -> Either UnpackageError ToSign
 getToSign Package{..} =
-  case decodeHex pkToSign of
-    Just hexDecoded ->
-      case fromVal @ToSign <$> unpackValue' hexDecoded of
-        Right toSign -> Right toSign
-        Left err -> Left $ UnpackFailure err
-    Nothing -> Left HexDecodingFailure
+  case fromVal @ToSign <$> unpackValue' (unHexJSONByteString pkToSign) of
+    Right toSign -> Right toSign
+    Left err -> Left $ UnpackFailure err
```

Comment:
I guess `first UnpackFailure` could be used here to make this definition shorter, but that's up to you.